cmd/run.go

```diff
@@ -49,20 +49,13 @@ var RunCommand = cli.Command{
 		heritageName := c.String("heritage-name")
 		detach := c.Bool("detach")
 		envVars := c.StringSlice("envvar")
-		envVarMap := make(map[string]string)
+		envVarMap, loadEnvVarMapErr := loadEnvVars(envName, heritageName)
+		if loadEnvVarMapErr != nil {
+			return cli.NewExitError(loadEnvVarMapErr.Error(), 1)
+		}
 		if len(envName) > 0 && len(heritageName) > 0 {
 			return cli.NewExitError("environment and heritage-name are exclusive", 1)
 		}
-		if len(envName) > 0 {
-			env, err := LoadEnvironment(envName)
-			if err != nil {
-				return cli.NewExitError(err.Error(), 1)
-			}
-			heritageName = env.Name
```

Comment:
Looks to me like we might still need this `heritageName = env.Name` part.

Later in the file, it does this
```go
api.DefaultClient.Post("/heritages/"+heritageName+"/oneoffs", bytes.NewBuffer(j))
```
which might not won't work right if `len(envName) > 0 && len(heritageName) == 0`.

Does that sound right?

Reply:
Hmm that code looks awfully suspicious. You are right. I'll try and fix it in a sane way